influx-sink.js

```diff
@@ -16,37 +23,102 @@ var defaults = {
   }
 }
 
-
-module.exports = function (opts) {
+module.exports = function (options) {
   var seneca = this
   var extend = seneca.util.deepextend
 
-  opts = extend(defaults, opts)
-  seneca.add({role: opts.role, hook: 'sink'}, handle_storage)
+  locals = extend(locals, options)
+
+  locals.log = bole(locals.plugin)
+  locals.client = influx(locals.influx)
+
+  locals.batch.list = {}
+  locals.batch.stream = through.obj(on_write)
```

Comment:
Instead of using through, you can just use a Writable and pass in a `https://nodejs.org/api/stream.html#stream_writable_writev_chunks_callback` function (you still need to implement `_write` though), so that you do not have a timeout, but you write down to influx as soon as your batch is filled up.